staging/src/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go

```diff
@@ -174,6 +174,8 @@ func (e Equalities) deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool,
 			}
 		}
 		return true
+	case reflect.String:
```

Comment:
A test case demonstrating what this is fixing would be good. We deep compare lots of things with strings already, it's unclear why this case is needed (and if it is needed, why cases for all primitive types are not also required)